core/src/test/java/org/apache/iceberg/rest/responses/TestDropNamespaceResponse.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.rest.RequestResponseTestBase;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestDropNamespaceResponse extends RequestResponseTestBase<DropNamespaceResponse> {
+
+  @Test
+  public void testRoundTripSerDe() throws JsonProcessingException {
+    assertRoundTripSerializesEquallyFrom(
+        "{\"dropped\":true}", DropNamespaceResponse.builder().dropped(true).build());
+
+    assertRoundTripSerializesEquallyFrom(
+        "{\"dropped\":false}", DropNamespaceResponse.builder().dropped(false).build());
+  }
+
+  @Test
+  public void testFailParsingWhenNullOrEmptyJson() {
+    String nullJson = null;
+    AssertHelpers.assertThrows("DropNamespaceResponse should fail to deserialize from a null JSON string",
+        IllegalArgumentException.class,
+        () ->  deserialize(nullJson));
+
+    String emptyString = "";
+    AssertHelpers.assertThrows("DropNamespaceResponse should fail to deserialize empty string as JSON",
+        JsonProcessingException.class,
+        () ->  deserialize(emptyString));
+
+    String emptyJson = "{}";
+    AssertHelpers.assertThrows(
+        "DropNamespaceResponse should fail to deserialize and validate if missing the field `dropped`",
+        IllegalArgumentException.class,
+        "Cannot instantiate a response to dropping a namespace with missing field(s): `dropped`",
+        () -> deserialize(emptyJson));
+  }
+
+  @Test
+  public void testFailWhenFieldsHaveInvalidValues() {
+    String invalidJsonWithStringType = "{\"dropped\":\"421\"}";
```

Comment:
What happens when the value is `null`?

Reply:
Added a test with null. It throws upon calling `validate`.